crates/core/src/network/handler/lists.rs

```diff
@@ -25,21 +28,76 @@ pub fn handle_get_authoring_list(get_list_data: GetListData, context: Arc<Contex
+type AddressVecMap = HashMap<EntryHash, Vec<AspectHash>>;
+fn convert_address_set_map(map: AddressSetMap) -> AddressVecMap {
+    let mut new_map = HashMap::new();
+    map.into_iter().for_each(|(entry, set)| {
+        let vec = set.into_iter().collect();
+        new_map.insert(entry, vec);
+    });
+    new_map
+}
+
+fn create_authoring_map(context: Arc<Context>) -> HashMap<EntryHash, HashSet<AspectHash>> {
+    let mut address_map: HashMap<EntryHash, HashSet<AspectHash>> = HashMap::new();
     for entry_address in get_all_public_chain_entries(context.clone()) {
+        // 1. For every public chain entry we definitely add the content aspect:
         let content_aspect = get_content_aspect(&entry_address, context.clone())
             .expect("Must be able to get content aspect of entry that is in our source chain");
-        address_map.insert(
-            EntryHash::from(entry_address.clone()),
-            vec![AspectHash::from(content_aspect.address())],
-        );
+
+        address_map
+            .entry(EntryHash::from(entry_address.clone()))
+            .or_insert_with(|| HashSet::new())
+            .insert(AspectHash::from(content_aspect.address()));
+
+        // 2. Then we might need to add a meta aspect as well depending on what kind of
+        //    entry this is.
+
+        // So we first unwrap the entry it self from the content aspect:
+        let (entry, header) = match content_aspect {
+            EntryAspect::Content(entry, header) => (entry, header),
+            _ => panic!("get_content_aspect must return only EntryAspect::Content"),
+        };
+
+        // And then we deduce the according base entry and meta aspect from that entry
+        // and its header:
+        let maybe_meta_aspect = match entry {
+            Entry::App(app_type, app_value) => {
+                header.link_update_delete().and_then(|updated_entry| {
+                    Some((
+                        updated_entry,
+                        EntryAspect::Update(Entry::App(app_type, app_value), header),
+                    ))
+                })
+            }
+            Entry::LinkAdd(link_data) => Some((
+                link_data.link.base().clone(),
+                EntryAspect::LinkAdd(link_data, header),
+            )),
+            Entry::LinkRemove((link_data, addresses)) => Some((
+                link_data.link.base().clone(),
+                EntryAspect::LinkRemove((link_data, addresses), header),
+            )),
+            Entry::Deletion(_) => Some((
+                header.link_update_delete().expect(""),
+                EntryAspect::Deletion(header),
+            )),
+            _ => None,
+        };
+
+        if let Some((base_address, meta_aspect)) = maybe_meta_aspect {
+            address_map
+                .entry(EntryHash::from(base_address.clone()))
+                .or_insert_with(|| HashSet::new())
+                .insert(AspectHash::from(meta_aspect.address()));
```

Comment:
We should hold on to the entry from before, when we added the content aspect. We are certain that the entry exists because we created it there. Then we can just insert. There should be no chance that the aspect list will only hold a meta entry and not the content entry, which this code currently implies.

Reply:
Nevermind, read this wrong, didn't see it's for a different entry